lib/Extractor/Candidates.cpp

```diff
@@ -818,7 +822,23 @@ void ExtractExprCandidates(Function &F, const LoopInfo *LI, DemandedBits *DB,
 
   for (auto &BB : F) {
     std::unique_ptr<BlockCandidateSet> BCS(new BlockCandidateSet);
+    BCS->BB = &BB;
     for (auto &I : BB) {
+      // Harvest Uses (Operands)
+      if (HarvestUses) {
+        for (auto &Op : I.operands()) {
+          if (auto U = dyn_cast<Instruction>(Op)){
```

Comment:
is this test necessary? what about a use that's an argument to a function, don't we want to optimize those too?

Reply:
The ExprBuilder only allows Instruction as input for now, I can make the ExprBuilder to allow argument as input and transform it into Souper::Var.

Reply:
This will be another PR.